x-pack/plugins/alerting/server/task_runner/transform_action_params.ts

```diff
@@ -98,6 +100,7 @@ export function transformActionParams({
       spaceId,
       tags,
       url: ruleUrl,
+      viewInAppUrl,
```

Comment:
Is there a way we could combine these two URLs? The rule URL currently defaults to the stack management page but the original issue was to allow rule types to customize that action variable. With this PR, `rule.url` is not customizable but the `rule.viewInAppUrl` is. What do you think of setting the `rule.url` to the `viewInAppUrl` if the `getViewInAppRelativeUrl` is defined and default to the stack management page otherwise? And remove this new context variable?

Reply:
I think that's a good idea, I went ahead and made the change in this commit https://github.com/elastic/kibana/pull/148671/commits/2f1138214be66bd142df52290cc068a8c850f91c and was able to reduce my code changes further in this commit https://github.com/elastic/kibana/pull/148671/commits/00e5b99cf99d8f586a78acf0ca54c2e7ec2d1de5.

Reply:
I also created a follow up discuss issue to see if we should match this URL behaviour in the email footer link => https://github.com/elastic/kibana/issues/151355.